yt/data_objects/profiles.py

```diff
@@ -1456,15 +1465,15 @@ def create_profile(
             o_bins.append(field_obin)
 
     args = [data_source]
-    for f, n, (mi, ma), l in zip(bin_fields, n_bins, ex, logs):
+    for f, n, (mi, ma), l in zip(bin_fields, n_bins, ex, logs, strict=True):
         if mi <= 0 and l:
             raise YTIllDefinedBounds(mi, ma)
         args += [f, n, mi, ma, l]
     kwargs = {"weight_field": weight_field}
     if cls is ParticleProfile:
         kwargs["deposition"] = deposition
     if override_bins is not None:
-        for o_bin, ax in zip(o_bins, ["x", "y", "z"]):
+        for o_bin, ax in zip(o_bins, ["x", "y", "z"], strict=False):
```

Comment:
This is one very case where it's apparent that `strict=False` is the intended behavior (found thanks to a test).

Reply:
Great catch.  Yes, I think this is assuming that it'll run out of axes based on the dimensionality.